### PR TITLE
Polyline refactoring

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/modeler/Object3d.py [hunk 1]
```diff
@@ -417,6 +417,37 @@ def __init__(self, object3d, edge_id):
         self._object3d = object3d
         self.oeditor = object3d.m_Editor
 
+    @property
+    def segment_info(self):
+        """Compute the segment info using the object oriented method (from AEDT 21R2 with beta options).
```
Comment (PipKat) [inline pyaedt/modeler/Object3d.py:422]:
```suggestion
        """Compute segment information using the object-oriented method (from AEDT 2021 R2 with beta options).
```

File: pyaedt/modeler/Object3d.py [hunk 2]
```diff
@@ -417,6 +417,37 @@ def __init__(self, object3d, edge_id):
...
+
+        Returns
+        -------
+            list
+                Segment info if available."""
```
Comment (PipKat) [inline pyaedt/modeler/Object3d.py:427]:
```suggestion
                Segment information if available.
    """
```

File: pyaedt/modeler/Object3d.py [hunk 3]
```diff
@@ -476,10 +507,9 @@ def length(self):
         >>> oEditor.GetVertexPosition
 
         """
-        if len(self.vertices) == 2:
-            length = GeometryOperators.points_distance(self.vertices[0].position, self.vertices[1].position)
-            return float(length)
-        else:
+        try:
+            return float(self.oeditor.GetEdgeLength(self.id))
+        except:
```
Comment (Alberto-DM) [inline pyaedt/modeler/Object3d.py:513]:
```suggestion
        except:  # pragma: no cover
```

**Review by Alberto-DM — APPROVED** [review]
LGTM. Great add!
Please check the suggested changes.

File: pyaedt/modeler/Object3d.py [hunk 4]
```diff
@@ -417,6 +417,37 @@ def __init__(self, object3d, edge_id):
...
+    def segment_info(self):
+        """Compute the segment info using the object oriented method (from AEDT 21R2 with beta options).
+
+        Returns
+        -------
+            list
+                Segment info if available."""
+        try:
+            self.oeditor.GetChildNames()
+        except:
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/modeler/Object3d.py]: added. Thanks

File: pyaedt/modeler/Object3d.py [hunk 5]
```diff
@@ -417,6 +417,37 @@ def __init__(self, object3d, edge_id):
...
+            return []
+        ll = list(self.oeditor.GetObjectsInGroup("Lines"))
+        self.oeditor.CreateObjectFromEdges(
+            ["NAME:Selections", "Selections:=", self._object3d.name, "NewPartsModelFlag:=", "NonModel#"],
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/modeler/Object3d.py]: Fixed.

File: pyaedt/modeler/Object3d.py [hunk 6]
```diff
@@ -417,6 +417,37 @@ def __init__(self, object3d, edge_id):
...
+        )
+        new_line = [i for i in list(self.oeditor.GetObjectsInGroup("Lines")) if i not in ll]
+        self.oeditor.GenerateHistory(
+            ["NAME:Selections", "Selections:=", new_line[0], "NewPartsModelFlag:=", "NonModel", "UseCurrentCS:=", True]
+        )
+        oo = self.oeditor.GetChildObject(new_line[0])
+        segment = {}
+        for prop in oo.GetChildObject(oo.GetChildNames()[0]).GetChildObject("Segment0").GetPropNames():
+            if "/" not in prop:
+                segment[prop] = oo.GetChildObject(oo.GetChildNames()[0]).GetChildObject("Segment0").GetPropValue(prop)
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/modeler/Object3d.py]: fixed

**Pull request merged**